ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -725,12 +726,13 @@ chainSyncClient mkPipelineDecision0 tracer cfg
     rollForward mkPipelineDecision n hdr theirTip
               = Stateful $ \kis -> traceException $ do
       now <- getMonotonicTime
-      -- Reject the block if invalid
       let hdrHash  = headerHash hdr
           hdrPoint = headerPoint hdr
-      isInvalidBlock <- atomically $ forgetFingerprint <$> getIsInvalidBlock
-      whenJust (isInvalidBlock hdrHash) $ \reason ->
-        disconnect $ InvalidBlock hdrPoint reason
+      unless (isPipeliningEnabled version) $ do
+        -- Reject the block if invalid
+        isInvalidBlock <- atomically $ forgetFingerprint <$> getIsInvalidBlock
+        whenJust (isInvalidBlock hdrHash) $ \reason ->
+          disconnect $ InvalidBlock hdrPoint reason
```

Comment:
We still check for _extending_ invalid chains though right?

Reply:
Yes, that happens here, in a background thread:

https://github.com/input-output-hk/ouroboros-network/blob/6d11459940a5d9a06a0109b75e56bc37a03cf940/ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs#L1013-L1017

Reply:
Hmm, I don't think it's reliable to use the `Watcher` for this.

Reply:
Update: While we don't think it is strictly necessary, I will go ahead and add a check that at this place in the code that ensures that the parent hash is not known-invalid.